runtime/include/http.h

```diff
@@ -30,46 +33,55 @@
 	"Server: SLEdge\r\n"       \
 	"Connection: close\r\n"    \
 	"Content-Type: %s\r\n"     \
-	"Content-Length: %s\r\n"   \
+	"Content-Length: %lu\r\n"  \
 	"\r\n"
 
 /* The sum of format specifier characters in the template above */
-#define HTTP_RESPONSE_200_TEMPLATE_FORMAT_SPECIFIER_LENGTH 4
+#define HTTP_RESPONSE_200_TEMPLATE_FORMAT_SPECIFIER_LENGTH 5
+
 
-/**
- * Calculates the number of bytes of the HTTP response containing the passed header values
- * @return total size in bytes
- */
-static inline size_t
-http_response_200_size(char *content_type, char *content_length)
+static inline int
+http_header_200_write(int fd, const char *content_type, size_t content_length)
 {
-	size_t size = 0;
-	size += strlen(HTTP_RESPONSE_200_TEMPLATE) - HTTP_RESPONSE_200_TEMPLATE_FORMAT_SPECIFIER_LENGTH;
-	size += strlen(content_type);
-	size += strlen(content_length);
-	return size;
+	return dprintf(fd, HTTP_RESPONSE_200_TEMPLATE, content_type, content_length);
```

Comment:
I was allocating a temporary in-memory buffer to sprintf to. This seemed silly when I realized I could dprintf directly instead. I forget about a bunch of these I/O functions, so it didn't occur to me until I skimmed APUE one night recently.